config/base/catalogd/manager/network_policy.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: controller-manager
+  namespace: system
+spec:
+  podSelector:
+    matchLabels:
+      control-plane: catalogd-controller-manager
+  policyTypes:
+    - Ingress
+    - Egress
```

Comment:
I think we jyst need Ingress right?

Reply:
First I did like you there, and I received this feedback from community: https://github.com/kubernetes-sigs/kubebuilder/pull/3853#discussion_r1685714616

So, we might need to reconsider.

Reply:
We explicitly want to allow all egress traffic both catalogd and operator-controller because they need to be able to communicate with unknown-in-advance image registries. If we are not explicitly granting that access, then another NetworkPolicy that matches our pod that is more restrictive could break us.

Reply:
I think we'll need to be more restrictive to make sense.
But I get your point. 
We can also enhance it on follow ups. So, 👍